File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [1] * n

        for i in range(n):
            for j in range(n):
                if i != j:
                    res[i] = res[i]*nums[j]
        return res                                       

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        List<int[]> list = new ArrayList<>();

        for(int num : nums){
            map.put(num , map.getOrDefault(num , 0) +1);
        }

        for(Map.Entry<Integer, Integer> entry : map.entrySet()) {
            list.add(new int[] {entry.getValue() , entry.getKey()});
        }

        list.sort((a , b) -> b[0] - a[0]);

        int[] res = new int[k];
        for ( int i =0 ; i < k ; i ++){
            res[i] =list.get(i)[1] ;
        }

        return res;
    }
}
